graphix/pauli.py

```diff
@@ -19,6 +21,58 @@ class IXYZ(enum.Enum):
     Z = 2
 
 
+class Sign(enum.Enum):
+    Plus = 1
+    Minus = -1
+
+    def __str__(self) -> str:
+        if self == Sign.Plus:
+            return "+"
+        return "-"
+
+    @staticmethod
+    def plus_if(b: bool) -> Sign:
+        if b:
+            return Sign.Plus
+        return Sign.Minus
+
+    @staticmethod
+    def minus_if(b: bool) -> Sign:
+        if b:
+            return Sign.Minus
+        return Sign.Plus
+
+    def __neg__(self) -> Sign:
+        return Sign.minus_if(self == Sign.Plus)
+
+    @typing.overload
+    def __mul__(self, other: Sign) -> Sign: ...
+
+    @typing.overload
+    def __mul__(self, other: Number) -> Number: ...
+
+    def __mul__(self, other):
+        if isinstance(other, Sign):
+            return Sign.plus_if(self == other)
+        if isinstance(other, Number):
+            return self.value * other
+        return NotImplemented
+
+    def __rmul__(self, other) -> Number | type(NotImplemented):
```

Comment:
`type(NotImplemented)` may be unnecessary as users never see it (results in exception if not implemented).

Reply:
Actually, `NotImplemented` may be seen if the user call `__rmul__` directly, but what I didn't know is that `NotImplemented` is an empty type for `mypy` (and `type(NotImplemented)` is just unsupported), so we never need to add explicitly `NotImplemented` in types.

Reply:
Fixed in 3884d45.